src/sinks/datadog/traces/request_builder.rs

```diff
@@ -271,11 +248,49 @@ impl DatadogTracesEncoder {
             _ => vec![],
         };
 
-        dd_proto::ApiTrace {
-            trace_id: trace_id as u64,
+        let chunk = dd_proto::TraceChunk {
+            priority: trace
+                .get("priority")
+                .and_then(|v| v.as_integer().map(|v| v as i32))
+                .unwrap_or(1i32),
```

Comment:
This field is supposed to always be there for traces coming from DD tracing lib, however the OLTP receiver in the `trace-agent` uses 1 (= [PriorityAutoKeep](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/sampler/sampler.go#L54-L55)) https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/api/otlp.go#L309. And that's mostly why I put this as the default value.

Reply:
Makes sense. Can we add that as a code comment? I think it will be useful for others who don't have as deep of a knowledge as you do with the DD Agent codebase.